tiledb/sm/array/array.cc

```diff
@@ -553,32 +553,55 @@ void Array::delete_array(const URI& uri) {
   throw_if_not_ok(this->close());
 }
 
-Status Array::delete_fragments(
+void Array::delete_fragments(
     const URI& uri, uint64_t timestamp_start, uint64_t timestamp_end) {
   // Check that query type is MODIFY_EXCLUSIVE
   if (query_type_ != QueryType::MODIFY_EXCLUSIVE) {
-    return LOG_STATUS(Status_ArrayError(
+    throw StatusException(Status_ArrayError(
         "[Array::delete_fragments] Query type must be MODIFY_EXCLUSIVE"));
   }
 
   // Check that array is open
   if (!is_open() && !controller().is_open(uri)) {
-    return LOG_STATUS(
+    throw StatusException(
         Status_ArrayError("[Array::delete_fragments] Array is closed"));
   }
 
   // Check that array is not in the process of opening or closing
   if (is_opening_or_closing_) {
-    return LOG_STATUS(Status_ArrayError(
+    throw StatusException(Status_ArrayError(
         "[Array::delete_fragments] "
         "May not perform simultaneous open or close operations."));
   }
 
   // Delete fragments
-  RETURN_NOT_OK(storage_manager_->delete_fragments(
-      uri.c_str(), timestamp_start, timestamp_end));
+  storage_manager_->delete_fragments(
+      uri.c_str(), timestamp_start, timestamp_end);
+}
 
-  return Status::Ok();
+void Array::delete_fragments_list(
+    const URI& uri, const std::vector<std::string> fragment_uris) {
+  // Check that query type is MODIFY_EXCLUSIVE
+  if (query_type_ != QueryType::MODIFY_EXCLUSIVE) {
```

Comment:
This is the same code as in Array::delete_fragments. Let's make a common function for this.